### PR TITLE
feat(ST-130): implement edit resource endpoint

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change
- Created dto for updateResourceDto
- Updated controller for implementing edit resource endpoint
- Updated resource service layer and repository

## Associated Ticket Link(s)
- https://trello.com/c/dyUaQ3TT

## Screenshots / Screen Recordings
N/A

### CODE REVIEW

File: src/resources/resources.service.ts [hunk 1]
```diff
@@ -58,4 +58,25 @@ export class ResourcesService {
...
+
+  async updateResource(
+    resourceId: number,
+    updateResourceDto: UpdateResourceDto,
+    file?: Express.Multer.File,
+  ) {
+    const resource = await this.resourcesRepository.findOneOrFail(resourceId);
+    if (file) {
+      const oldFileKey = resource.fileUrl.split("project-dev-bucket/")[1];
+      await this.fileUploadsService.deleteFromS3(oldFileKey);
```
Comment (ishmam-iqbal-sazim) [inline src/resources/resources.service.ts:70]: Probably best to have checks here for error cases

Comment (ghost) [reply src/resources/resources.service.ts:70]: Already addressed the issue on the previous PR. Used try catch block on the fileUpload.service layer

**Review by ragib-hossain-sazim — APPROVED** [review]

File: src/resources/resources.controller.ts [hunk 2]
```diff
@@ -36,4 +38,14 @@ export class ResourcesController {
...
   }
+
+  @Put(":classroomId/resources/:resourceId")
+  @UseInterceptors(FileInterceptor("file"))
+  updateResource(
+    @Param("resourceId") resourceId: number,
+    @Body() updateResourceDto: UpdateResourceDto,
+    @UploadedFile() file?: Express.Multer.File,
+  ) {
+    return this.resourcesService.updateResource(resourceId, updateResourceDto, file);
```
Comment (ishmam-iqbal-sazim) [inline src/resources/resources.controller.ts:49]:
```suggestion
    return this.resourcesService.updateOne(resourceId, updateResourceDto, file);
```
[NIT]

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment

**Pull request merged**